src/solver/conv_asm_implicit_gemm_gtc_fwd.cpp

```diff
@@ -0,0 +1,557 @@
+        {"fwd", "fp32", 32, 0, 4, 64, 16, 4, 64, 1, 1, 1, 1, {1, 1, 1, 1}, {1, 16, 1, 4}, {1, 8, 2, 1}, {1, 2, 1, 32}, 0}, 
+        {"fwd", "fp32", 4, 0, 4, 64, 16, 4, 64, 1, 1, 1, 1, {1, 1, 1, 1}, {1, 16, 1, 4}, {1, 4, 4, 1}, {1, 4, 1, 16}, 0}, 
+        {"fwd", "fp32", 4, 1, 4, 64, 16, 4, 64, 1, 1, 1, 1, {1, 1, 1, 1}, {1, 16, 1, 4}, {1, 4, 4, 1}, {1, 4, 1, 16}, 0}, 
+        {"fwd", "fp32", 1, 0, 4, 64, 16, 4, 64, 1, 1, 1, 1, {1, 1, 1, 1}, {1, 16, 1, 4}, {1, 4, 4, 1}, {1, 4, 1, 16}, 0}, 
+        {"fwd", "fp32", 1, 1, 4, 64, 16, 4, 64, 1, 1, 1, 1, {1, 1, 1, 1}, {1, 16, 1, 4}, {1, 4, 4, 1}, {1, 4, 1, 16}, 0}, 
+        {"fwd", "fp32", 4, 0, 4, 64, 16, 4, 64, 1, 1, 1, 1, {1, 1, 1, 1}, {1, 16, 1, 4}, {1, 1, 16, 1}, {1, 16, 1, 4}, 0}, 
+        {"fwd", "fp32", 4, 1, 4, 64, 16, 4, 64, 1, 1, 1, 1, {1, 1, 1, 1}, {1, 16, 1, 4}, {1, 1, 16, 1}, {1, 16, 1, 4}, 0}, 
+        {"fwd", "fp32", 1, 0, 4, 64, 16, 4, 64, 1, 1, 1, 1, {1, 1, 1, 1}, {1, 16, 1, 4}, {1, 1, 16, 1}, {1, 16, 1, 4}, 0}, 
+        {"fwd", "fp32", 1, 1, 4, 64, 16, 4, 64, 1, 1, 1, 1, {1, 1, 1, 1}, {1, 16, 1, 4}, {1, 1, 16, 1}, {1, 16, 1, 4}, 0}, 
+        {"fwd", "fp32", 16, 0, 16, 16, 4, 16, 16, 1, 1, 1, 1, {1, 1, 1, 1}, {1, 4, 1, 16}, {1, 1, 1, 1}, {1, 4, 1, 16}, 0}, 
+        {"fwd", "fp32", 4, 0, 64, 4, 16, 64, 4, 1, 1, 1, 1, {1, 1, 16, 1}, {1, 16, 1, 4}, {1, 1, 1, 1}, {1, 16, 1, 4}, 0}, 
+        {"fwd", "fp32", 4, 1, 64, 4, 16, 64, 4, 1, 1, 1, 1, {1, 1, 16, 1}, {1, 16, 1, 4}, {1, 1, 1, 1}, {1, 16, 1, 4}, 0}, 
+        {"fwd", "fp32", 1, 0, 64, 4, 16, 64, 4, 1, 1, 1, 1, {1, 1, 16, 1}, {1, 16, 1, 4}, {1, 1, 1, 1}, {1, 16, 1, 4}, 0}, 
+        {"fwd", "fp32", 1, 1, 64, 4, 16, 64, 4, 1, 1, 1, 1, {1, 1, 16, 1}, {1, 16, 1, 4}, {1, 1, 1, 1}, {1, 16, 1, 4}, 0}, 
+    };
+
+    return kernel_param_list;
+}
+
+/*
+* Return with true if valid configuration is found, and its kernel_name (optional), block_size, grid_size as well.
+* Return with false if no valid kernel configuration can be selected for the specified ConvolutionContext
+*/
+static inline bool FindImplicitGemmGtcDynamicFwdKernel(const ConvolutionContext& ctx, std::string *p_kernel_name, int *p_block_size, int *p_grid_size)
+{
+    auto tunables  = GetImplicitGemmGtcDynamicFwdXdlopsTunablesList();
+    int n          = ctx.batch_sz;
+    int c          = ctx.n_inputs;
+    int k          = ctx.n_outputs;
+    int ho         = ctx.out_height;
+    int wo         = ctx.out_width;
+    int stride_h   = ctx.out_height > 1 ? ctx.kernel_stride_h : 1;
+    int stride_w   = ctx.out_width > 1 ? ctx.kernel_stride_w : 1;
+    int dilation_h = ctx.kernel_size_h > 1 ? ctx.kernel_dilation_h : 1;
+    int dilation_w = ctx.kernel_size_w > 1 ? ctx.kernel_dilation_w : 1;
+    int pad_h      = ctx.pad_h;
+    int pad_w      = ctx.pad_w;
+    int y          = ctx.kernel_size_h;
+    int x          = ctx.kernel_size_w;
+
+    int gemm_m = k;
+    int gemm_n = n * ho * wo;
+    int gemm_k = c * y * x; 
+
+    auto pConfig   = tunables.begin();
```

Comment:
`pConfig` should be more suitblable in `for( auto pConfig ....`, or clang-tidy might not pass?

Reply:
Changed !